app/src/main/java/org/kiwix/kiwixmobile/localFileTransfer/LocalFileTransferActivity.kt

```diff
@@ -0,0 +1,382 @@
+import kotlinx.android.synthetic.main.activity_local_file_transfer.text_view_empty_peer_list
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.utils.AlertDialogShower
+import org.kiwix.kiwixmobile.core.utils.KiwixDialog
+import org.kiwix.kiwixmobile.kiwixActivityComponent
+import org.kiwix.kiwixmobile.localFileTransfer.WifiDirectManager.Companion.getDeviceStatus
+import org.kiwix.kiwixmobile.localFileTransfer.adapter.WifiP2pDelegate
+import org.kiwix.kiwixmobile.localFileTransfer.adapter.WifiPeerListAdapter
+import java.util.ArrayList
+import javax.inject.Inject
+
+/**
+ * Created by @Aditya-Sood as a part of GSoC 2019.
+ *
+ * This activity is the starting point for the module used for sharing zims between devices.
+ *
+ * The module is used for transferring ZIM files from one device to another, from within the
+ * app. Two devices are connected to each other using WiFi Direct, followed by file transfer.
+ *
+ * File transfer involves two phases:
+ * 1) Handshake with the selected peer device, using [PeerGroupHandshakeAsyncTask]
+ * 2) After handshake, starting the files transfer using [SenderDeviceAsyncTask] on the sender
+ * device and [ReceiverDeviceAsyncTask] files receiving device
+ */
+@SuppressLint("GoogleAppIndexingApiWarning", "Registered")
+class LocalFileTransferActivity : BaseActivity(),
+  WifiDirectManager.Callbacks {
+  @Inject
+  lateinit var alertDialogShower: AlertDialogShower
+
+  @Inject
+  lateinit var wifiDirectManager: WifiDirectManager
+
+  @Inject
+  lateinit var locationManager: LocationManager
+
+  private var isFileSender = false // Whether the device is the file sender or not
+  private var filesForTransfer = ArrayList<FileItem>()
+  private var fileListAdapter: FileListAdapter? = null
+  private var wifiPeerListAdapter: WifiPeerListAdapter? = null
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_local_file_transfer)
+
+    /*
+     * Presence of file Uris decides whether the device with the activity open is a sender or receiver:
+     * - On the sender device, this activity is started from the app chooser post selection
+     * of files to share in the Library
+     * - On the receiver device, the activity is started directly from within the 'Get Content'
+     * activity, without any file Uris
+     * */
+    val filesIntent = intent
+    val fileUriArrayList: ArrayList<Uri>?
+    fileUriArrayList = filesIntent.getParcelableArrayListExtra(Intent.EXTRA_STREAM)
+    isFileSender = fileUriArrayList?.isNotEmpty() == true && fileUriArrayList.size > 0
+
+    val toolbar: Toolbar =
+      findViewById(R.id.toolbar)
+    // I know this is not the best method, but android extensions were showing 'unknown reference' error
```

Comment:
Prefer to leave comments like this on a draft PR, not in the source code.

This is fine KAE doesn't work properly across modules

Reply:
Sir, KAE=?

Reply:
Kotlin Android Extensions, https://antonioleiva.com/kotlin-android-extensions/

They have been somewhat de-emphasised by the Kotlin team because of the release of ViewBinding by Google (or that is my theory), which is a damn shame.
But also we don't need to find views anymore once Compose is released so I would expect both of these libraries to disappear fully, the `Parcelize` functionality was great though

Reply:
Yep, i also read about it that dataBinding is better than kotlinSynthetics. Thanks for the info though

Reply:
https://stackoverflow.com/questions/58351239/viewbinding-vs-kotlin-android-extensions-with-synthetic-views sort of, I tend to agree with this answer but it all really doesn't matter once compose gets a beta release

Reply:
I'm also too excited for Compose. Just out of curiosity, any plan/idea do we have for migration to hilt? (i know its sounds stupid though, but still :sweat_smile: )

Reply:
I mentioned hilt today on a ticket about a crash with our component not being initialised, I am all for Dagger being made usable